settlement-pipelines/src/bin/init_settlement.rs

```diff
@@ -134,6 +137,9 @@ async fn real_main(reporting: &mut ReportHandler<InitSettlementReport>) -> anyho
     )
     .await?;
 
+    // waiting for data finalization on-chain
+    sleep(std::time::Duration::from_secs(8)).await;
```

Comment:
maybe set it as a constant? also what's the reason behind 8 seconds? usually it takes that much?

Reply:
Yeah, the constant is an arbitrary value that helped with getting data from RPC nodes more reliably. I consider it as "a workaround" and in general I don't like it. But I haven't got time to verify if switching to finalized finality could for example help or to get some more respectful solution.

I placed the value under a constant as you suggested.

WDYT, could that be?

Reply:
the difference between 8s and block finality is not that big since a finalized block takes 32 x 400ms = 12.8s, but we can try if we see this workaround is not enough.